core/src/main/resources/xtdb/sql/parser/sql.ebnf

```diff
@@ -501,7 +501,7 @@ with_or_without_time_zone
     ;
 
 interval_type
-    : 'INTERVAL' interval_qualifier
+    : 'INTERVAL' [ interval_qualifier ]
```

Comment:
Making `interval_qualifier` optional here, such that we can use `CAST(... AS INTERVAL)`